docs/architecture/dapr-for-net-developers/dapr-at-20000-feet.md

```diff
@@ -0,0 +1,257 @@
+
+The blue boxes across the center of the figure represent the DAPR building blocks. Each abstracts a distributed application capability that your application can consume.
+
+The bottom row highlights the portability of Dapr and the diverse environments across which it can run.
+
+## Dapr architecture
+
+At this point, our jet turns around and flies back over Dapr, descending in altitude, giving us a closer look at how Dapr works.
+
+### Building blocks
+
+From our new perspective, we have a more detailed view of the Dapr **building blocks**.
+
+A building block encapsulates a distributed application capability. You can access the functionality through the HTTP or gRPC APIs that we saw earlier. Figure 2-2 shows the available blocks for Dapr v 1.0.
+
+![Dapr building blocks](./media/building-blocks.png)
+
+**Figure 2-2**. Dapr building blocks.
+
+The following table describes the services provided by each block.
+
+| Building Block | Description |
+| :-------- | :-------- |
+| [Service-to-service invocation](https://github.com/dapr/components-contrib/tree/master/nameresolution) | Invoke direct, secure service-to-service calls using platform agnostic protocols and well-known endpoints. |
+| [Publish and subscribe](https://github.com/dapr/components-contrib/tree/master/state) | Implement secure, scalable pub/sub messaging between services. |
+| [State management](https://github.com/dapr/components-contrib/tree/master/pubsub) | Support contextual information for long running stateful services. |
+| [Observability](https://github.com/dapr/components-contrib/tree/master/bindings) | Monitor and measure message calls across networked services. |
+| [Secrets](https://github.com/dapr/components-contrib/tree/master/middleware) | Securely access external secret stores. |
+| [Actors](https://github.com/dapr/components-contrib/tree/master/secretstores) | Encapsulate logic and data in reusable actor objects. |
+| [Resource bindings and triggers](https://github.com/dapr/components-contrib/tree/master/exporters) | Trigger code from events raised by external resources with bi-directional communication. |
+
+Building blocks abstract the implementation of distributed application capabilities from your services. Figure 2-3 shows this interaction.
+
+![Dapr building blocks](./media/building-block-integration.png)
+
+**Figure 2-3**. Dapr building block integration.
+
+Under the hood, building blocks invoke Dapr components that provide the concrete implementation for each resource. The code for your service is only aware of the building block. It takes no dependencies on external SDKs or libraries - Dapr handles the plumbing for you. Each building block is independent. You can use one, some, or all of them in your application. As a value-add, Dapr building blocks bake in industry best practices.
```

Comment:
```suggestion
Building blocks invoke Dapr components that provide the concrete implementation for each resource. The code for your service is only aware of the building block. It takes no dependencies on external SDKs or libraries - Dapr handles the plumbing for you. Each building block is independent. You can use one, some, or all of them in your application. As a value-add, Dapr building blocks bake in industry best practices.
```

The term "Under the hood" is over utilized... try to avoid it.